src/GraphQL/PersistedDocuments/InvalidRequestError.cs

```diff
@@ -0,0 +1,14 @@
+using GraphQL.Execution;
+
+namespace GraphQL.PersistedDocuments;
+
+/// <summary>
+/// Represents an error that occurs when a request contains both query and documentId parameters.
+/// </summary>
+public class InvalidRequestError : RequestError
+{
+    /// <inheritdoc cref="InvalidRequestError"/>
+    public InvalidRequestError() : base("The request must not have both query and documentId parameters.")
```

Comment:
I'm not sure this is correct
> Note: When persisting a document for which the identifier has been derived by the client, it is generally good practice for the client to issue both the GraphQL Document and the document identifier to the server; the server could then regenerate the document identifier from the GraphQL Document independently, and check that the identifiers match before storing the Document.

Reply:
Where is that from? Are you sure you’re not reading docs on APQ support?

Reply:
The last paragraph under [Persisting a Document](https://github.com/graphql/graphql-over-http/blob/52d56fb36d51c17e08a920510a23bdc2f6a720be/spec/Appendix%20A%20--%20Persisted%20Documents.md)

Reply:
Here: https://github.com/graphql/graphql-over-http/pull/264/files#diff-9be5577e05ae2112d2b8f95584b162d0dec01453bf6c85df58bf5db4f2c9727aR122-R129

Reply:
I believe this relates to the preceding note, discussing "automatic persisted queries (APQ)" support, where the client knows the text of the request and determines a hash for it, sending it instead of or addition to the original request.  We support this with the `UseAutomaticPersistedQueries` method, supporting Apollo's specification listed [here](https://www.apollographql.com/docs/apollo-server/performance/apq/), which differs from this specification.

Unfortunately the current spec does not provide enough information to make a working APQ implementation.  Without specifying the result in the case of a failed match against a hash, the client is unable to conclusively identify the failure reason and resend the request with the query string.  As such, both the preceding comment and this one have no purpose at this time.

Reply:
Understood. Then all looks good